src/RageSoundMixBuffer.cpp

```diff
@@ -4,17 +4,29 @@
 
 #include <cmath>
 #include <cstdint>
+#include <cstdlib>
 
 RageSoundMixBuffer::RageSoundMixBuffer()
 {
-	m_iBufSize = m_iBufUsed = 0;
-	m_pMixbuf = nullptr;
+	// See how many samples we can stuff into 2MB.
+	size_t BUF_SIZE = 2 * 1024 * 1024 / sizeof(float);
+	
+	m_pMixbuf = static_cast<float*>(std::malloc(BUF_SIZE * sizeof(float)));
+	if (m_pMixbuf == nullptr) {
+		ASSERT_M(false, "Failed to allocate memory for the sound mixing buffer");
+	} else {
```

Comment:
No need for the else, since ASSERT_M will close the program anyways. It adds a block level for not much gain here.

Reply:
Ahh true. Okay, removed the else statement here.